cmd/prometheus/main.go

```diff
@@ -344,6 +345,10 @@ func main() {
 	klog.SetLogger(log.With(logger, "component", "k8s_client_runtime"))
 
 	level.Info(logger).Log("msg", "Starting Prometheus", "version", version.Info())
+	if bits.UintSize < 64 {
+		level.Warn(logger).Log("msg", "This Prometheus binary hasn't been compiled for a 64-bit architecture. Due to memory mapping constraints of 32-bit systems, it is highly recommended to switch to a 64-bit build of Prometheus.", "GOARCH", runtime.GOARCH)
```

Comment:
It's a virtual memory constraint rather than an mmap constraint. mmap itself works fine on 32bit.